### ISSUE TITLE
Discovery: Legacy Dashboard: Grade Downloads

### ISSUE THREAD

**Issue by pdpinch** [mitocw/edx-platform#75]
I believe all the legacy dashboard features in "Grade Downloads" have been ported to the new dashboard. 

See, for example, https://staging.mitx.mit.edu/courses/DevOps/0.001/2015_Summer/legacy_instructor_dash
- [X] "Dump List of Enrolled Students" => "Data Download" / "List enrolled students' Profile information"
- [x] "Dump All Raw Grades for all students in this course" => "Student Admin" / "View Gradebook" ?
- [x] "Download CSV of all RAW Grades" => 
- [ ] "Download CSV of answer distributions" => "Data Download" / "Generate Problem Grade Report" ?

![screenshot-staging mitx mit edu 2015-08-21 11-44-20](https://cloud.githubusercontent.com/assets/430126/9412502/0d4aabe2-47fa-11e5-9b60-d39cd641048f.png)

**Comment by sarina** [comment]
Downloading raw/csv grades may need to be ported, but it would be good to understand what the use cases are and whether or not it's really necessary to port the info, or if existing structures can be used (eg Insights).

CSV of answer distributions is terribly unperformant and doesn't use the same analytics structures that we have in place. "Porting" this should be an entirely new implementation that goes through Celery and uses analytics APIs.

**Comment by gsidebo** [comment]
@pdpinch A couple questions:

1. My assumption right now is that the first button in that screenshot already has some equivalent in the current release, and that the latter 3 buttons _probably_ need to be ported over, but there could possibly be some equivalent in the current release. Is that correct?
1. Do we care about performance for these buttons? I was deliberate about not spending too much time trying to optimize or find better-performing equivalents to the functionality I ported over in the ["Export Grades" PR](https://github.com/mitodl/edx-platform/pull/7). If we care enough, I can take that time to look around the codebase (including "Insights" and the analytics APIs, which Sarina mentions but I haven't reviewed yet).

**Comment by gsidebo** [comment]
A few observations after digging through the codebase:

- The RAW grade report, as it existed in Dogwood, seems to be a breakdown of enrolled students' grades for individual problems in a course. There is a rough equivalent in ficus - the CSV generated by "Generate Problem Grade Report" in the ["Data Download" tab](https://mitx-qa-draft-next.mitx.mit.edu/courses/DevOps/0.001/2015_Summer/instructor#view-data_download
). Here are the two reports on "DevOps: 0.001 Ops" for comparison: [Problem Grade Report](https://github.com/mitocw/edx-platform/files/1061880/gen_problem_grade_report_qa.xlsx) - [Legacy RAW Grades Report](https://github.com/mitocw/edx-platform/files/1061881/legacy_RAW_grades.xlsx). A couple differences:
  1. The problem grade report represent each problem with 2 columns (one for "Possible" points and one for "Earned") and the column headers are much more verbose.
  1. The legacy report has one extra column: "SGA Unit 1". I'm not sure what that is, but I'd be more inclined to trust the current problem grade report to accurately reflect all of the existing problems and their grades. 
**TLDR;** We should decide if the problem grade report is a good enough replacement for the legacy RAW grade report (in my opinion, that looks like a 'yes')
- The "Dump All Raw Grades" button just spits out that same CSV report in the browser window, so our implementation of that will depend on our decision about the RAW grade CSV functionality.
- There doesn't seem to be any equivalent to the answer distribution report in ficus. Here's an example report from QA: [legacy_answer_distibutions.xlsx](https://github.com/mitocw/edx-platform/files/1061928/legacy_answer_distibutions.xlsx). It looks like it's a breakdown of every unique answer to every problem in the course with the number of times that specific answer value was submitted. It looks like a decent chunk of work to reimplement this, so I think it's well worth our time to figure out if anyone still needs this and what their use cases are

I'm moving this back to 'Ready' until we make some decisions about the above

**Comment by pdpinch** [comment]
I think I'm going to have to consult with the ed tech team on this and get back to you.

**Comment by pdpinch** [comment]
Thanks Gavin. 

I agree that "Dump All Raw Grades for all students in this course" is close enough to  "Generate Problem Grade Report"

And I've confirmed that the data available from "Download CSV of answer distributions" is equivalent to what is accessible from analytics.odl.mit.edu.
